cpp/src/arrow/array.cc

```diff
@@ -1334,7 +1334,7 @@ class NullArrayFactory {
     }
 
     template <typename T>
-    enable_if_base_list<T, Status> Visit(const T&) {
+    enable_if_var_length_list<T, Status> Visit(const T&) {
```

Comment:
Nit: "var size" rather than "var length".
(we have FixedSizeList, which has a variable-sized length ;-)

Reply:
done.